community: Add maritalk streaming (sync and async)

rodrigo-f-nogueira opened the pull request.

rodrigo-f-nogueira committed:
add stream

rodrigo-f-nogueira committed:
fix typo

rodrigo-f-nogueira committed:
add streaming to the documentation

rodrigo-f-nogueira committed:
fix example

rodrigo-f-nogueira committed:
fix lint

rodrigo-f-nogueira committed:
fix lint

ccurme commented on libs/community/langchain_community/chat_models/maritalk.py:
```diff
@@ -1,10 +1,25 @@
-from typing import Any, Dict, List, Optional, Union
+from typing import Any, AsyncIterator, Dict, List, Optional, Union
 
+import httpx
```
Since `httpx` is an [optional dependency](https://github.com/langchain-ai/langchain/blob/40f846e65da37a1c00d72da9ea64ebb0f295b016/libs/community/pyproject.toml#L92) of langchain-community, we should ensure users can import this code without the dependency installed with no side effects. See [contributing guide](https://python.langchain.com/docs/contributing/code#working-with-optional-dependencies). Consider moving this import inside the class (e.g., as is done [here](https://github.com/langchain-ai/langchain/blob/40f846e65da37a1c00d72da9ea64ebb0f295b016/libs/community/langchain_community/llms/chatglm3.py#L76)).

ccurme commented on libs/community/langchain_community/chat_models/maritalk.py:
```diff
@@ -163,6 +178,72 @@ def _call(
+                "POST",
+                "https://chat.maritaca.ai/api/chat/inference", 
+                data=json.dumps(data),
+                headers=headers
```
There are some formatting errors here. See [contributing guide](https://python.langchain.com/docs/contributing/code) for how to resolve-- e.g., you may need to run `make format_diff` from /libs/community/.

rodrigo-f-nogueira committed:
Merge branch 'langchain-ai:master' into add-maritalk-stream

rodrigo-f-nogueira committed:
fix import

rodrigo-f-nogueira committed:
Merge branch 'add-maritalk-stream' of https://github.com/maritaca-ai/langchain into add-maritalk-stream

rodrigo-f-nogueira committed:
fix import

rodrigo-f-nogueira committed:
fix lint

rodrigo-f-nogueira committed:
aa

rodrigo-f-nogueira committed:
fix import

rodrigo-f-nogueira committed:
aaa

ccurme commented on libs/community/langchain_community/chat_models/maritalk.py:
```diff
@@ -163,6 +178,75 @@ def _call(
+        run_manager: Optional[AsyncCallbackManagerForLLMRun] = None,
+        **kwargs: Any,
+    ) -> ChatResult:
+        if self.streaming:
```
@rodrigo-f-nogueira do you intend for `.streaming` to be an attribute of `ChatMaritalk`? if so you need to add it to the class definition.

rodrigo-f-nogueira replied:
Hi @ccurme , this should be fixed now

rodrigo-f-nogueira committed:
fix lint

rodrigo-f-nogueira committed:
Merge branch 'master' into add-maritalk-stream

RosevalJr committed:
Refactor(maritalk): add import error for httpx dependency

RosevalJr committed:
Refactor(maritalk): add streaming functionality

- Add stream method
- Standardize error handling
- Change parent class for ChatMaritalk

RosevalJr committed:
Refactor(maritalk): fix streaming examples

RosevalJr committed:
Merge branch 'master' into add-maritalk-stream

RosevalJr committed:
Fix(maritalk): Change url for api endpoint

RosevalJr commented:
Hello @ccurme.

Sorry for the delay on this PR. The raised issues have been resolved. Can we proceed with the merge?

RosevalJr committed:
Fix(maritalk): apply lint

RosevalJr committed:
Merge branch 'master' into add-maritalk-stream

ccurme commented:
Thanks @RosevalJr, looks good! It looks like maintainers don't have permissions to edit the PR. Could you run `make format` from the monorepo root as described [here](https://python.langchain.com/docs/contributing/documentation/technical_logistics/#linting-and-formatting)? This will fix the formatting in the notebook.

RosevalJr committed:
Fix(maritalk): apply lint

RosevalJr committed:
Merge branch 'master' into add-maritalk-stream

RosevalJr commented:
Thanks, @ccurme. 

The code is now linted.

RosevalJr committed:
Fix(maritalk): apply lint
